### PR TITLE
Butter Fee Converter

### PR DESCRIPTION
**Pull request by anthonymartin**
- Add `ButterFeeConverter` and `IButterFeeConverter`. This contract claims the Set streaming fee (in BTR) and any accrued redemption fees (in 3CRV) for Butter, redeems BTR for 3CRV, withdraws to a single stablecoin, pays a keeper tip, and forwards the balance to the `RewardsManager`
- Add and update interfaces: `IStreamingFeeModule`, `IContractRegistry`
- Add a deployment script
- Update Forge fork tests to use latest `forge-std`

### CODE REVIEW

**Comment by bhanusanghi** [comment]
Notes from the latest updated state of the PR. 

- Exposes mainly 2 keeper incentivised functions for 
1. accrueFee - This accrues fee in Butter Tokens from streamingFeeModule and deposits it into butterBatchProcessor to be converted to Butter.
2. transferConvertedFeesToRewardsManager -  This checks if the batch is processed and redeems all the processed batches. Then converts the recvd 3crv to USDC and transfers it to the rewardsManager.

Missing from this PR ->
- Need to uncomment the updatePrefferedStableCoinIndex function.
- Need to add KeeperIncentivized modifiers to the contract.
- Need to add tests to confirm is the keeper recvs the incentive.
- When redeeming multiple batches at once, check if there is any inefficiency.

**Comment by ecmendenhall** [comment]
This should handle redemption fees from ButterBatchProcessing, too.

**Comment by RedVeil** [comment]
Very nice! Small and clean contract. I quite like it.

File: packages/hardhat/contracts/core/defi/butter/ButterFeeConverter.sol [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    keeperTip = _keeperTip;
+    setApprovals();
+  }
+
+  /**
+   * @notice accrue fee from Set streaming fee module in BTR and deposit it into the current redeem batch
+   * @dev The stableCoinIndex_ must align with the index in the curve three-pool
+   */
+  function accrueFee() external override keeperIncentive(0) {
+    require(streamingFeeModule.getFee(butter) > 0, "No Fee accrued yet");
```
[the remark this answers is not in the document]
Comment (anthonymartin) [reply packages/hardhat/contracts/core/defi/butter/ButterFeeConverter.sol:83]: streaming fees are always collected. and so long as there's tvl, this should always return true. I would say we can even remove the statement. 

the idea for this contract is to collect _all_ the fees

[the remark this answers is not in the document]
Comment (RedVeil) [reply packages/hardhat/contracts/core/defi/butter/ButterFeeConverter.sol:83]: okidok ill remove it

File: .vscode/settings.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "editor.formatOnSave": true
   },
   "[solidity]": {
-    "editor.defaultFormatter": "NomicFoundation.hardhat-solidity"
+    "editor.defaultFormatter": "JuanBlanco.solidity"
```
Comment (anthonymartin) [inline .vscode/settings.json:32]: @RedVeil was this intentional?

**Pull request merged**